examples/README.md

```diff
@@ -0,0 +1,24 @@
+# Mongooose Network Library Examples
+
+In order to build and run any of the existing examples, please follow:
+https://mongoose.ws/documentation/#how-to-build-and-run-examples
+
+## Contributing
+
+Rules for creating a new example:
+
+- Makefile golden reference for desktop/server example: [http-server](http-server)
+- Makefile golden reference for the embedded example:
+   [stm32/nucleo-f746zg-baremetal](stm32/nucleo-f746zg-baremetal)
+- An example must build on Windows, Mac and Ubuntu Linux systems
+- Assume that user installed tools according to https://mongoose.ws/tutorials/tools/
+- Makefile must not include any other make files
+- Use `CFLAGS` for compilation options
+- Add `CFLAGS_EXTRA` to the end of `CFLAGS`, to allow user to specify
```

Comment:
We've been using EXTRA_CFLAGS so far. Anyway, I think CFLAGS_EXTRA is more appropriate